x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/servicenow/helpers.ts

```diff
@@ -6,7 +6,46 @@
  */
 
 import { EuiSelectOption } from '@elastic/eui';
-import { Choice } from './types';
+import {
+  ENABLE_NEW_SN_ITSM_CONNECTOR,
+  ENABLE_NEW_SN_SIR_CONNECTOR,
+  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
+} from '../../../../../../actions/server/constants/connectors';
+import { IErrorObject } from '../../../../../public/types';
+import { AppInfo, Choice, RESTApiError, ServiceNowActionConnector } from './types';
 
 export const choicesToEuiOptions = (choices: Choice[]): EuiSelectOption[] =>
   choices.map((choice) => ({ value: choice.value, text: choice.label }));
+
+export const isRESTApiError = (res: AppInfo | RESTApiError): res is RESTApiError =>
+  (res as RESTApiError).error != null || (res as RESTApiError).status === 'failure';
+
+export const isFieldInvalid = (
+  field: string | undefined,
+  error: string | IErrorObject | string[]
+): boolean => error !== undefined && error.length > 0 && field !== undefined;
```

Comment:
Yes, it will always be an array here. I agree that the type is misleading but I need to match the alerting's framework types. Although (`undefined > 0 === false`).